### ISSUE TITLE
Flutter compute and network requests

### ISSUE THREAD

**Issue by 3xcellentTuber** [flutter/flutter#48183]
I am building an application that has its own high level protocol that runs over TCP, thus I am using RawSocket and RawServerSocket to handle TCP communications. I have a static class that handles the communications and the logic of which message should I respond to first (Used various messaging queues to handle that). With flutter being Single Threaded, in order to not hang the main UI I created a Peridic Timer that checks if there are new messages in the message queue every 2.5 seconds (the messages enter the queue thanks to the event listener of the socket), then I handle the messages and the UI seems to run smooth.

I had the feeling that if I were to send a large amount of data such as an image or a song, I would hang up the Event looper due to it reading from the socket. So I tried to create a callback function to handle read/write events and call it up using the compute method so that it'll run on its own Isolate and return me the data once finished.

Unfortunately, I encountered the exception : `flutter Illegal argument in isolate message object extends Native Wrapper...`

This is obviously caused due to me attempting to send the RawSocket as it is, if I were to be able to encode it to a json string or some binary representation of it and then decode it inside the callback function this would work, but I don't seem to find an actual way to establish that.

My question is: How can I somehow send the socket over to the callback function while using compute? And, is the Read/Write operation of the socket managed in its own Isolate ?, because if they are, then I really don't need to use compute at all. I was unable to find information about this, apart from flutter stating that compute can send primative types only. Creating an Isolate and applying complex logic in it's send/recv methods seems like an overkill to establish such a simple thing. I really don't understand how the Flutter team excpects us to create applications that run smoothly with one thread (yes, they support async operations, but that is still managed on one single thread which makes large network operations really painful, that is if the RawSocket Read/Write operations are NOT run on a different Isolate)

## Steps to Reproduce
1. create a callback function
2. call the compute function and pass the callback and a map that includes a socket or just a socket

```
Uint8List readFromTcpSocketCallback(Map<String, dynamic> m){
  return (m[ConstantValues.tcpSocketDefinitionForMap.toString()] as RawSocket).read(m[ConstantValues.amountReadDefinitionForMap.toString()]);
}

class Communications{
...
static Future<Uint8List> _readAllFromTcpSocket(RawSocket socket, int length)async{
    Uint8List finalBufferReceived;
    Uint8List buffer;
    int amountRead = 0;
    do
    {
      buffer = await compute(readFromTcpSocketCallback, {ConstantValues.tcpSocketDefinitionForMap.toString() : socket, 
      ConstantValues.amountReadDefinitionForMap.toString() : length - amountRead});
      amountRead += buffer.length;
      finalBufferReceived.isEmpty ? finalBufferReceived = buffer : finalBufferReceived.addAll(buffer);
    }
    while(amountRead < length);
    return finalBufferReceived;
  }
}
```

## Logs
```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.535], locale en-IL)
    • Flutter version 1.12.13+hotfix.5 at C:\flutter
    • Framework revision 27321ebbad (4 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\eitan\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[!] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.41.1)
    • VS Code at C:\Users\eitan\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.7.1

[√] Connected device (1 available)
    • GM1900 • ******** • android-arm64 • Android 10 (API 29)
```

**Comment by goderbauer** [comment]
This sounds like a question that should be asked on one of our other community channels, e.g. StackOverflow. See https://flutter.dev/community for more information.

**Comment by 3xcellentTuber** [comment]
Hey,
I did not get a proper solution from SO. I'd like to reopen that here or get further info in order to redesign my project.
I am currently trying to understand if I even need to be worried about this situation:
I realized that there are serveral main Isolates in dart.
Are RawSocket and RawServerSocket operations blocking the main UI Isolate or are the Read/Write operations being performed on a different Isolate (I/O maybe?)
If they are blocking - for example if I READ 50 GB right now would the UI thread be hung?
if it does, I don't understand why it isn't pointed out, flutter stands to be a fast and reactive framework, which I believe it is, but if that is the situation why don't we have an option to pass sockets to Isolates??
Thanks in advance.
